backend/src/libs/packages/encrypt/base-encrypt.package.ts

```diff
@@ -0,0 +1,17 @@
+import bcrypt from 'bcryptjs';
+
+import { config } from '#libs/packages/config/config.js';
+
+import { type Encrypt } from './libs/types/types.js';
+
+class BaseEncrypt implements Encrypt {
+  public async generateSalt(rounds?: number): Promise<string> {
+    return await bcrypt.genSalt(rounds ?? config.ENV.ENCRYPT.NUMBER_OF_ROUNDS);
```

Comment:
We don't want to access config here
```suggestion
    return await bcrypt.genSalt(rounds);
```

Reply:
Do you mean to access config at all in BaseEncrypt  logic, or just directrly in function?

Reply:
Or access it in user.repository and pass to generateSalt(). The field rounds in generateSalt() is required, so I need to pass here some value.

Reply:
If prop rounds is required, then there is no needed to add default value from config)

Reply:
I just wanted to clarify,where in such cases is better to declare a constant with rounds value. Just create constant in user.repository? I implemented it in config because I thought that it may be used in other parts of the project and createing common default value is ok

Reply:
You need to pass config to user service and pass salt value when you call generateSalt method

Reply:
And the reason why I made `rounds `param in `generateSalt `function optional is that `genSalt ` function in `bcrypt ` has rorunds optional param with defallt = 10